test/mersenne.spec.ts

```diff
@@ -2,51 +2,55 @@ import { beforeAll, beforeEach, describe, expect, it } from 'vitest';
 import type { Mersenne } from '../src/internal/mersenne/mersenne';
 import mersenneFn from '../src/internal/mersenne/mersenne';
 import { seededRuns } from './support/seededRuns';
+import { times } from './support/times';
 
 const NON_SEEDED_BASED_RUN = 25;
 
 describe('mersenne twister', () => {
   const mersenne: Mersenne = mersenneFn();
 
-  for (const seed of [...seededRuns, [42, 1, 2], [1337, 1, 2], [1211, 1, 2]]) {
-    describe(`seed: ${JSON.stringify(seed)}`, () => {
-      beforeEach(() => {
-        mersenne.seed(seed);
-      });
+  describe.each(
+    (seededRuns as Array<number | number[]>)
+      .concat([[42, 1, 2]], [[1337, 1, 2]], [[1211, 1, 2]])
+      .map((s) => [s])
+  )('seed: %j', (seed) => {
```

Comment:
```suggestion
  describe.each(
    [...seededRuns, [42, 1, 2], [1337, 1, 2], [1211, 1, 2]].map((v) => [v])
  )('seed: %j', (seed) => {
```

Reply:
The `.map((v) => [v])` is only needed due to some inconsistency in the handling of %j vs the args,

````txt
stdout | test/mersenne.spec.ts > mersenne twister > seed: 1337 > should return deterministic value for next()
seed:  1337

stdout | test/mersenne.spec.ts > mersenne twister > seed: 42 > should return deterministic value for next()
seed:  [ 42, 1, 2 ]

stdout | test/mersenne.spec.ts > mersenne twister > seed: 1211 > should return deterministic value for next()
seed:  [ 1211, 1, 2 ]

stdout | test/mersenne.spec.ts > mersenne twister > seed: 1211 > should return deterministic value for next()
seed:  1211

stdout | test/mersenne.spec.ts > mersenne twister > seed: 1337 > should return deterministic value for next()
seed:  [ 1337, 1, 2 ]

stdout | test/mersenne.spec.ts > mersenne twister > seed: 42 > should return deterministic value for next()
seed:  42
````

Reply:
IMO we should report that to vitest as a bug.

Reply:
The `.map((v) => [v])` is only needed due to some inconsistency we otherwise have in the handling of %j vs the args,

````txt
stdout | test/mersenne.spec.ts > mersenne twister > seed: 1337 > should return deterministic value for next()
seed:  1337

stdout | test/mersenne.spec.ts > mersenne twister > seed: 42 > should return deterministic value for next()
seed:  [ 42, 1, 2 ]

stdout | test/mersenne.spec.ts > mersenne twister > seed: 1211 > should return deterministic value for next()
seed:  [ 1211, 1, 2 ]

stdout | test/mersenne.spec.ts > mersenne twister > seed: 1211 > should return deterministic value for next()
seed:  1211

stdout | test/mersenne.spec.ts > mersenne twister > seed: 1337 > should return deterministic value for next()
seed:  [ 1337, 1, 2 ]

stdout | test/mersenne.spec.ts > mersenne twister > seed: 42 > should return deterministic value for next()
seed:  42
````

IMO we should report that to vitest as a bug.

Reply:
I'm not sure if this is a bug upstream 🤔 
vitest expect an array of "tuple"s that are spreaded as args.

So `[1337]` becomes `(1337) => {}` but `[1337, 1, 2]` would become `(1337, 1, 2) => {}`
But we want `([1337, 1, 2]) => {}` instead, so we need to wrap them in a tuple with only one first value.

and then `.each` takes an array of tuples so we would have:

```ts
describe.each([
  [42],
  [1337],
  [1211],
  [[42, 1, 2]],
  [[1337, 1, 2]],
  [[1211, 1, 2]],
])...
```

Do you think we can somehow improve the code on our side to achieve the wanted result in a better way?

Reply:
> vitest expect an array of "tuple"s that are spreaded as args.

Then why doesnt it throw a type error if called otherwise?

> Do you think we can somehow improve the code on our side to achieve the wanted result in a better way?

IMO my suggestion pretty much handles this. We could shorten it further by mapping the arrays as well. :man_shrugging: 

```suggestion
  describe.each(
    [...seededRuns, ...seededRuns.map((v) => [v, 1, 2])].map((v) => [v])
  )('seed: %j', (seed) => {
```

Any maybe extract the `.map((v) => [v])` part in a separate method called tuplize or similar.